### PR TITLE
Integrate CI tests with github actions

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
This PR implements github actions to run tests including
- check_format
- catkin build local planner

github actions seem to start the ci container much faster compared to travis.
The check format test was removed from travis to speed up travis builds

**Additional context**
- catkin build on global planner requires https://github.com/PX4/containers/pull/207 to be merged
- Added clang format to px4/containers so that we don't need to install it every time we run checks in https://github.com/PX4/containers/pull/221

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        cd $HOME/catkin_ws/src
+        ln -s $GITHUB_WORKSPACE
+        catkin config --cmake-args -DCMAKE_BUILD_TYPE=Release -DCATKIN_ENABLE_TESTING=False -DDISABLE_SIMULATION=ON
+        catkin build -j$(nproc) -l$(nproc) local_planner
+    - name:
+      working-directory:
+      run: |
+        cd $HOME/catkin_ws/src
+        catkin config --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCATKIN_ENABLE_TESTING=True
+        catkin build avoidance local_planner --no-deps -v -i --catkin-make-args tests
```
Comment (jkflying) [inline .github/workflows/build_test.yml]: This is building the  tests, but doesn't run them

Comment (Jaeyoung-Lim) [reply .github/workflows/build_test.yml]: @jkflying I am waiting for valgrind to be included in the container

Comment (jkflying) [reply .github/workflows/build_test.yml]: Ok, we can still run the tests, just not in valgrind though.

**Comment by jkflying** [comment]
> /github/home/catkin_ws/devel/lib/avoidance/avoidance-test: error while loading shared libraries: libxmlrpcpp.so: cannot open shared object file: No such file or directory

Looks like there is a missing dependency

**Comment by Jaeyoung-Lim** [comment]
@jkflying The issue is coming from environment sourcing

**Comment by Jaeyoung-Lim** [comment]
@TSC21 Any ideas on why it can't find ros dependencies?

**Comment by TSC21** [comment]
> @TSC21 Any ideas on why it can't find ros dependencies?

Have you tried `source /opt/ros/melodic/setup.bash`?

**Comment by Jaeyoung-Lim** [comment]
@Jaeyoung-Lim source doesn't seem to exist, but should this be part of `catkin_ws/devel/setup.bash` I will give it a try though

File: .github/workflows/build_test.yml (hunk 1 above)
Comment (Jaeyoung-Lim) [reply .github/workflows/build_test.yml]: Tests now running :)

**Comment by Jaeyoung-Lim** [comment]
@jkflying Tests are now running

**Comment by jkflying** [comment]
Is there anything missing on the Github Actions builds?

(I guess the coverage builds?)

I'd prefer to drop the Travis builds.

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    - uses: actions/checkout@v1
+      with:
+        token: ${{ secrets.ACCESS_TOKEN }}
+    - name: covereage_checks
+      working-directory:
+      run: |
+        apt update
+        apt install -y lcov rubygems ruby-dev
+        gem install coveralls-lcov
+        tools/generate_coverage.sh && coveralls-lcov repo_total.info
```
Comment (jkflying) [inline .github/workflows/build_test.yml]: The coverage also runs the functional tests, so you need to start `roscore`

Comment (Jaeyoung-Lim) [reply .github/workflows/build_test.yml]: Ah, okay. but how is it done in https://github.com/PX4/avoidance/blob/master/.travis.yml#L89 ?

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        (roscore &) && for f in ~/catkin_ws/devel/lib/*/*-test-roscore; do valgrind --leak-check=full --track-origins=yes --error-exitcode=1 --suppressions=$HOME/catkin_ws/src/avoidance/local_planner/test/valgrind_suppressions.sup $f || status=1; done && (if (( $status > 0 )); then exit $status; fi)
+      shell: bash
+    - name: coverage
+      working-directory:
+      run: |
+        apt update
+        apt install -y lcov rubygems ruby-dev ros-kinetic-octomap-*
+        gem install coveralls-lcov
+        cd $HOME/catkin_ws/src/avoidance
+        tools/generate_coverage.sh
```
Comment (jkflying) [inline .github/workflows/build_test.yml]: I think you need a 
```suggestion
        (roscore &) && tools/generate_coverage.sh
```

**Pull request closed without merging**